File: Aufgabe1/Quadratwurzel/sqrt.hs
{-# OPTIONS_GHC -Wno-unrecognised-pragmas #-}
{-# HLINT ignore "Use camelCase" #-}
import System.IO

persion = 0.000001
isGoodEnough x y = abs (x*x - y) < persion
improve x y = 0.5*(y + x/y)

newton_sqrt :: Double -> Double
newton_sqrt x = iterate x 1
  where iterate x y | isGoodEnough x y =  y
                    | otherwise = iterate x (improve x y )

main :: IO ()
main = putStr "Bitte geben Sie eine Zahl ein: " >>
  hFlush stdout >>
  getLine >>=
  (\xStr ->
    let x = read xStr :: Double
    in putStr ("Die Wurzel aus " ++ xStr ++ " ist: ") >> print (newton_sqrt x))

File: Aufgabe1/Quadratwurzel/sqrt.rs

fn newton_sqrt(x : f32) -> f32 {
    let percision = 0.0000001;
    let mut xn = 1.0;
    while (xn * xn - x).abs() > percision {
        xn = 0.5*(xn + x/xn);
    }
    return xn;
}
fn main(){
    let mut line = String::new();
    println!("Bitte gib ein Zahl ein ");
    std::io::stdin().read_line(&mut line).unwrap();
    println!("{}" ,line);
    let num = line.trim().parse::<f32>().expect("Keine Zahl");
    println!("Die Wurzel aus {} ist: {}", num , newton_sqrt(num));
}